tests/src/python/test_provider_postgres.py

```diff
@@ -2362,6 +2362,65 @@ def testSkipConstraintCheck(self):
                 0, QgsFieldConstraints.Constraint.ConstraintUnique, 59
             )
         )
+        self.assertTrue(
+            vl.dataProvider().skipConstraintCheck(
+                0,
+                QgsFieldConstraints.Constraint.ConstraintUnique,
+                QgsUnsetAttributeValue(),
+            )
+        )
+        self.assertTrue(
+            vl.dataProvider().skipConstraintCheck(
+                0,
+                QgsFieldConstraints.Constraint.ConstraintNotNull,
+                QgsUnsetAttributeValue(),
+            )
+        )
```

Comment:
To be safe, can you add a test here for a field index which DOESN'T have a default clause? Those fields should return False to skipConstraintCheck if passed a unset attribute value (at least for the ConstraintNotNull constraint)

Reply:
@nyalldawson ok, done, test passes locally, let's hope CI is benevolent today.